Sources/ScipioKit/Cache/CacheSystem.swift

```diff
@@ -94,35 +95,38 @@ public struct CacheKey: Hashable, Codable, Equatable {
 
 public protocol CacheStorage {
     func existsValidCache(for cacheKey: CacheKey) async -> Bool
-    func fetchArtifacts(for cacheKey: CacheKey, to destinationDir: AbsolutePath) async throws
-    func cacheFramework(_ frameworkPath: AbsolutePath, for cacheKey: CacheKey) async
+    func fetchArtifacts(for cacheKey: CacheKey, to destinationDir: URL) async throws
+    func cacheFramework(_ frameworkPath: URL, for cacheKey: CacheKey) async
 }
 
 struct CacheSystem {
     private let rootPackage: Package
     private let buildOptions: BuildOptions
-    private let outputDirectory: AbsolutePath
+    private let outputDirectory: URL
     private let storage: (any CacheStorage)?
     private let fileSystem: any FileSystem
 
     enum Error: LocalizedError {
         case revisionNotDetected(String)
         case compilerVersionNotDetected
+        case couldNotReadVersionFile(URL)
 
         var errorDescription: String? {
             switch self {
             case .revisionNotDetected(let packageName):
                 return "Repository version is not detected for \(packageName)."
             case .compilerVersionNotDetected:
                 return "Compiler version not detected. Please check your environment"
+            case .couldNotReadVersionFile(let path):
+                return "Could not read VersionFile \(path.path)"
```

Comment:
Error is thrown here https://github.com/giginet/Scipio/pull/9/files#diff-c95016fdd23c7e500c6ca14a1de27966f249347c35a9e51c9e9974bde9d69d94R161 but will be converted to `return false`. So I'm curious how this error message works well.

Reply:
This error might not raise. So I do not care about it.

Reply:
i see